trie/trie.go

```diff
@@ -434,6 +434,9 @@ func (t *Trie) resolveHash(n hashNode, prefix []byte) (node, error) {
 
 	hash := common.BytesToHash(n)
 
+	if node := t.db.node(hash, t.cachegen); node != nil {
+		return node, nil
+	}
 	enc, err := t.db.Node(hash)
```

Comment:
You can call `t.db.diskdb.Get(hash)` here to avoid taking the DB lock twice.

Reply:
t.db.Node can pull up RLP blobs from the cache, some that might not yet have been written to the database.

Reply:
But you are calling `t.db.node` just above, that also pulls from the cache right?

Reply:
It doesn't do parsing, but I agree it's a bit convoluted, I'll trink about solving this nicer.

Reply:
Done. PTAL